src/main/java/dev/j3fftw/litexpansion/items/WateringCan.java

```diff
@@ -0,0 +1,163 @@
+package dev.j3fftw.litexpansion.items;
+
+import dev.j3fftw.litexpansion.Items;
+import dev.j3fftw.litexpansion.LiteXpansion;
+import dev.j3fftw.litexpansion.utils.Utils;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import me.mrCookieSlime.Slimefun.cscorelib2.chat.ChatColors;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.TreeType;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.data.Ageable;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockFertilizeEvent;
+import org.bukkit.event.player.PlayerBucketEvent;
+import org.bukkit.event.world.StructureGrowEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.material.Cauldron;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.String.valueOf;
+
+public class WateringCan extends SimpleSlimefunItem<ItemUseHandler> {
+
+    private static final int USE_INDEX = 7;
+
+    public WateringCan() {
+        super(Items.LITEXPANSION, Items.WATERING_CAN, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            Items.REFINED_IRON, null, Items.REFINED_IRON,
+            Items.REFINED_IRON, null, Items.REFINED_IRON,
+            null, Items.REFINED_IRON, null
+        });
+    }
+
+    @Override
+    public ItemUseHandler getItemHandler() {
+        return e -> {
+            e.cancel();
+            Optional<Block> bOpt = e.getClickedBlock();
+
+            if (bOpt.isPresent()) {
+                Block b = bOpt.get();
+                ItemStack item = e.getItem();
+                Player p = e.getPlayer();
+                BlockData blockData = b.getBlockData();
+                ItemMeta meta = item.getItemMeta();
+                List<String> lore = meta.getLore();
+
+                // Filling
+                if (b.getType() == Material.CAULDRON) {
+
+                    if (blockData.getAsString().contains("level=3")) {
```

Comment:
Nonononono. Use the Levelled blockdata interface.
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/block/data/Levelled.html